### PR TITLE
feat: add useZoom hook

### PR DESCRIPTION
**Pull request by hwbllmnn**
@terrestris/devs Please review.

### CODE REVIEW

File: src/hooks/useZoom.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * active state will not get changed.
+ * @param constructor returns an interaction to be added to the map, will be called again, if the interaction needs
+ * to be updated
+ * @param active
+ */
+export const useZoom = (
+  /**
+   * Whether the zoom in shall be animated.
+   */
+  animate: boolean,
```
Comment (ahennr) [inline src/hooks/useZoom.ts:20]: Would a custom type e.g. `UseZoomProperties` having default values make sense here?

Comment (hwbllmnn) [reply src/hooks/useZoom.ts]: Yes, extracted it to a `MapState` type and changed it to an object.

File: src/hooks/useZoom.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import {
+  useEffect
+} from 'react';
+
+import useMap from './useMap';
+
+/**
+ * This hook adds an interaction to the map and removes/updates it if the dependency array changes.
+ * It accepts an optional active parameter that toggles the active state of the interaction. If it is undefined the
+ * active state will not get changed.
```
[the remark this answers is not in the document]
Comment (hwbllmnn) [reply src/hooks/useZoom.ts]: Right, classic copy/paste mistake, corrected it.

**Review by ahennr — APPROVED** [review]

**Review by simonseyock — COMMENTED** [review]
I was thinking if this functionality would better be a util function? Or return a `zoomTo` function from the hook if you want to wrap the `useMap` call.

A function would be a little more flexible. If I had for example a callback where I want to change the zoom level, I would need a state variable if I want to use this hook. A function I could call directly.

If I had a state variable that I wanted to react to, it would still easibly be possible. I could do something like this:
```
useEffect(
  () => zoomTo(map, { extent }),
  [extent]
);
```

**Comment by simonseyock** [comment]
Also if the dependency changes unexpectedly at one point it will immediately reset the zoom of the map. (this is minor and should be avoidable through clean programming)

On another note, if the zoom for example should be reset to the same value it was set to before, this will not update, because the value that the hooks knows of has not changed. This would be rather annoying.

I think the main problem I have is, that this hook does not have a two-way databinding to the openlayers map. So this hook is not aware of any changes in the map. It will always just set the zoom to a certain level. This does not feel right for me.

I think we should not attempt to create a two-way databinding here, even though it might be possible. I think it might cause many rerenders in react if we are not very careful.

**Review by simonseyock — CHANGES_REQUESTED** [review]
I'd prefer it if this was a function. But I am open to arguments :)

**Comment by hwbllmnn** [comment]
> I was thinking if this functionality would better be a util function? Or return a `zoomTo` function from the hook if you want to wrap the `useMap` call.
> 
> A function would be a little more flexible. If I had for example a callback where I want to change the zoom level, I would need a state variable if I want to use this hook. A function I could call directly.
> 
> If I had a state variable that I wanted to react to, it would still easibly be possible. I could do something like this:
> 
> ```
> useEffect(
>   () => zoomTo(map, { extent }),
>   [extent]
> );
> ```

Alright, I changed it to a util function.

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**